pgjdbc/src/test/java/org/postgresql/test/jdbc2/ResultSetTest.java

```diff
@@ -1152,4 +1161,47 @@ private Map<String, Integer> getResultSetColumnNameIndexMap(ResultSet stmt) {
     return null;
   }
 
+  private static class CallableWithConnection implements Callable<Boolean> {
+
+    private final Connection connection;
+    private final int expectedYear;
+
+    protected CallableWithConnection(Connection connection, int expectedYear) {
+      this.connection = connection;
+      this.expectedYear = expectedYear;
+    }
+
+    @Override
+    public Boolean call() throws SQLException {
+      try (Statement statement = connection.createStatement()) {
+        for (int i = 0; i < 10; i++) {
+          try (ResultSet resultSet = statement.executeQuery(
+              String.format("SELECT unnest(array_fill('8/10/%d'::timestamp, ARRAY[%d]))",
+                  expectedYear, 500))) {
+            while (resultSet.next()) {
+              Timestamp d = resultSet.getTimestamp(1);
+              int year = 1900 + d.getYear();
+              if (year != expectedYear) {
+                return false;
+              }
+            }
+          }
+        }
+      }
+      return true;
+    }
+
+  }
+
+  @Test
+  public void testTimestamp() throws InterruptedException, ExecutionException, TimeoutException {
+    ExecutorService e = Executors.newFixedThreadPool(2);
+    Future<Boolean> future1 = e.submit(new CallableWithConnection(con, 7777));
+    Future<Boolean> future2 = e.submit(new CallableWithConnection(con, 2017));
+    assertTrue(future1.get(1, TimeUnit.MINUTES));
+    assertTrue(future2.get(1, TimeUnit.MINUTES));
```

Comment:
The failure would be challenging to understand: `expected true got false`.
assertEquals + message would make the test easier to maintain

Reply:
done